### PR TITLE
Fix nullptr dereferences

### PR DESCRIPTION
**Pull request by Crunsher**
This prevents a number of nullptr dereferences within icinga 2.
Refs CVE-2018-6534

### CODE REVIEW

**Comment by Crunsher** [comment]
I have no idea why commit c508d74d4 is in there :/

**Comment by dnsmichi** [comment]
Wrong rebase base, I'd suggest to start with a fresh master and cherry-pick on-top into a new branch.

File: lib/base/object.hpp [hunk 1]
```diff
@@ -57,6 +57,10 @@ extern Value Empty;
 	DECLARE_PTR_TYPEDEFS(klass); \
 	IMPL_TYPE_LOOKUP();
 
+#define RequireNotNull(ptr) RequireNotNullInternal(ptr, #ptr)
```
Comment (dnsmichi) [inline lib/base/object.hpp]: Macros should be capitalised throughout, so `REQUIRE_NOT_NULL` would be correct, wouldn't it?

**Comment by Crunsher** [comment]
When merging back into the master the files relevant to the commit are not touched. Sometimes git works in mysterious ways :roll_eyes:

File: lib/base/object.hpp (hunk 1 above)
Comment (Crunsher) [reply lib/base/object.hpp]: @N-o-X ^

**Pull request merged**